test/lpi2Spec.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const crypto = require('crypto')
+const uuid = require('uuid4')
+const ilpPacket = require('ilp-packet')
+const btpPacket = require('btp-packet')
+const base64url = require('base64url')
+
+const chai = require('chai')
+chai.use(require('chai-as-promised'))
+const assert = chai.assert
+
+const ObjStore = require('./helpers/objStore')
+const PluginPaymentChannel = require('..')
+const MockSocket = require('./helpers/mockSocket')
+
+const info = {
+  currencyCode: 'USD',
+  currencyScale: 2
+}
+
+const options = {
+  currencyCode: 'USD',
+  currencyScale: 2,
+  maxBalance: '1000000',
+  minBalance: '-40',
+  server: 'btp+wss://user:placeholder@example.com/rpc',
+  info: info
+}
+
+describe.only('LPI2', () => {
+  beforeEach(async function () {
+    options._store = new ObjStore()
+    this.plugin = new PluginPaymentChannel(options)
+
+    this.mockSocketIndex = 0
+    this.mockSocket = new MockSocket()
+    this.mockSocket
+      .reply(btpPacket.TYPE_MESSAGE, ({ requestId }) => btpPacket.serializeResponse(requestId, []))
+
+    await this.plugin.addSocket(this.mockSocket, { username: 'user', token: 'placeholder' })
+    await this.plugin.connect()
+
+    this.error = {
+      code: 'F00',
+      name: 'Bad Request',
+      triggeredAt: new Date(),
+      data: JSON.stringify({ message: 'Peer isn\'t feeling like it.' })
+    }
+
+    const expiry = new Date(Date.now() + 10000)
+    this.fulfillment = 'gHJ2QeIZpstXaGZVCSq4d3vkrMSChNYKriefys3KMtI'
+    this.condition = base64url(crypto
+      .createHash('sha256')
+      .update(this.fulfillment, 'base64')
+      .digest())
+
+    this.transfer = {
+      id: uuid(),
```

Comment:
Isn't the role of transaction id taken over by the promise that sendTransfer returns?

Reply:
Good point. We can still keep it in BTP (to keep this compatible with current BTP) but the LPI has no reason to expose it.